Makefile.am

```diff
@@ -25,6 +25,12 @@ EXTRA_DIST += META AUTHORS COPYRIGHT LICENSE NEWS NOTICE README.md
 
 @CODE_COVERAGE_RULES@
 
+.PHONY: gitrev
+gitrev:
+	-${top_srcdir}/scripts/make_gitrev.sh
+
+BUILT_SOURCES = gitrev
```

Comment:
I think you should add a call to `-${top_srcdir}/scripts/make_gitrev.sh` in `dist-hook:`.  This target is run at the end of `make dist` and it let's you add additional files prior to the `tar`.  Right now it's used to ensure an updated META file is included in the the final tarball.  This would ensure that tarball also includes the git hash if available, and a valid version of `sys/zfs_gitrev.h`.

Reply:
I added a dependency on `gitrev`, which I think will do the same thing without duplicating code.  Let me know if you see a problem with this (I'm not a makefile expert).

Reply:
Well, it does the same thing as what you suggested.  However, zfs_gitrev.h doesn't end up in the tarball.  I suspect that we would need to tell it to generate the file in an alternate location.

Reply:
Turns out we copy the .c/.h files before doing `dist-hook`.  So I had to also add a line to dist-hook to copy zfs_gitrev.h into the distdir.  Now I have:
```
$ make dist
...
make[2]: Entering directory '/export/home/delphix/zfs'
./scripts/make_gitrev.sh
cp ./include/zfs_gitrev.h zfs-0.8.0/include; \
sed -i 's/Release:[[:print:]]*/Release:      rc1/' \
	zfs-0.8.0/META
...
$ tar -tzf zfs-0.8.0.tar.gz |grep git
zfs-0.8.0/scripts/make_gitrev.sh
zfs-0.8.0/include/zfs_gitrev.h
```